### PR TITLE
DlgContrast Update

### PR DESCRIPTION
**Pull request by Lunalo**
Almost done

### CODE REVIEW

**Comment by Lunalo** [comment]
@africanmathsinitiative/developers 

This is ready for review

**Comment by maxwellfundi** [comment]
@dannyparsons  this is ready fro your review

File: instat/dlgContrasts.vb [hunk 1]
```diff
@@ -38,15 +39,25 @@ Public Class dlgContrasts
...
+        bReset = False
+        autoTranslate(Me)
         TestOKEnabled()
     End Sub
 
+    Private Sub SetRCodeforControls(bReset As Boolean)
+        ucrSelectorForContrast.SetRCode(clsSetContrast, bReset)
+        ucrReceiverForContrasts.SetRCode(clsSetContrast, bReset)
+        ucrInputContrastName.SetRCode(clsSetContrast, bReset)
+    End Sub
```
Comment (lilyclements) [inline instat/dlgContrasts.vb]: How come you are setting these controls for the same class separately and not doing the overall
SetRCode(Me, ucrbase..., bReset) line?

**Comment by lilyclements** [comment]
Also, as a general discussion point: I can make my factor a contrast, then I can convert it to numeric. Is this what we want? I don't know much about contrasts

**Comment by lilyclements** [comment]
@dannyparsons as long as we're okay for me to convert a factor to a numeric and for it to hold its contrast status, this is ready to review

**Comment by dannyparsons** [comment]
Is this the general issue of receivers not clearing after converting?

**Comment by lilyclements** [comment]
No, sorry. If you define contrasts, it only allows you to for factors. Then it comes up in the metadata of what the contrast is defined as. When you change from a factor to numeric, the metadata remains that the contrast is defined as that. It confused me since this dialog doesn't allow you to define contrasts for a numeric, but the software allows a contrast to be defined for a numeric.

**Comment by dannyparsons** [comment]
Although it is useful if you then convert it back to factor the contrasts are not lost

**Comment by Lunalo** [comment]
@lilyclements 

What do you think about this ?

**Comment by lilyclements** [comment]
Sorry, yes that sounds reasonable to me. I was just checking there wasn't another dialog where having a numeric-contrast would mess things up.

@dannyparsons this is ready to review

**Pull request merged**